sphinx/builders/linkcheck.py

```diff
@@ -40,10 +40,11 @@
 
 uri_re = re.compile('([a-z]+:)?//')  # matches to foo:// and // (a protocol relative URL)
 
-Hyperlink = NamedTuple('Hyperlink', (('next_check', float),
-                                     ('uri', Optional[str]),
+Hyperlink = NamedTuple('Hyperlink', (('uri', Optional[str]),
                                      ('docname', Optional[str]),
                                      ('lineno', Optional[int])))
+CheckRequest = NamedTuple('CheckRequest', (('hyperlink', Optional[Hyperlink]),
+                                           ('next_check', float)))
```

Comment:
Since the check requests are used in the `PriorityQueue`, the comparison should first be on `next_check`, then `hyperlink`. It ensures they get the highest priority hyperlink (lower next_check) first.
```suggestion
CheckRequest = NamedTuple('CheckRequest', (('next_check', float))),
                                           ('hyperlink', Optional[Hyperlink]))
```

Reply:
I see.